specification/ml/_types/TrainedModel.ts

```diff
@@ -75,7 +75,7 @@ export class TrainedModelDeploymentStats {
    * The deployment stats for each node that currently has the model allocated.
    * In serverless, stats are reported for a single unnamed virtual node.
    */
-  nodes: TrainedModelDeploymentNodesStats
+  nodes: TrainedModelDeploymentNodesStats[]
```

Comment:
Sounds good! I did this change based on the response I get and the type that appears in the docs https://www.elastic.co/guide/en/elasticsearch/reference/current/get-trained-models-stats.html

Reply:
This `T | T[]` (single-or-many) is a common pattern for ES responses. It might be just `T[]` in this case, but we have to check the server code to be sure.

Reply:
I've found the line of server code that adds the nodes field, I'm not familiar with this codebase but looks like it's always an array https://github.com/elastic/elasticsearch/blob/d0253a6c7b76b74965ad88ad835bc75eb2b2041e/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/assignment/AssignmentStats.java#L636

Reply:
Thanks for checking! Yes, this indeed should be always an array.